recipes/web.rb

```diff
@@ -0,0 +1,27 @@
+package 'st2web'
+
+node.default['nginx']['default_site_enabled'] = false
+
+# Ensure directory for Nginx SSL certificates is available
+directory '/etc/ssl/st2' do
+  owner 'root'
+  group 'root'
+  mode '0755'
+  action :create
+end
+
+openssl_x509 '/etc/ssl/st2/st2.crt' do
+  common_name node.name
+  org 'StackStorm'
+  org_unit 'Information Technology'
+  country 'US'
+end
+
+include_recipe 'chef_nginx::repo'
+include_recipe 'chef_nginx::default'
+
+nginx_site 'st2.conf' do
+  template 'nginx/st2.conf.erb'
```

Comment:
If I understand correctly the code of the `nginx_site` resource, that is already taken care of: https://github.com/chef-cookbooks/chef_nginx/blob/v4.0.1/resources/site.rb#L42-L57